### PR TITLE
[#4364] feat(iceberg): Support GCS storage for Iceberg REST server

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
Support GCS storage for Iceberg REST server

### Why are the changes needed?

Fix: #4364

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
1. start Iceberg REST server with following config:
```
gravitino.iceberg-rest.warehouse = gs://xxx/test
gravitino.iceberg-rest.io-impl= org.apache.iceberg.gcp.gcs.GCSFileIO
```
2. run spark sqls to create Iceberg table

### CODE REVIEW

**Comment by FANNG1** [comment]
@jerryshao @yuqi1129 @diqiu50 please help to review when you are free, thanks

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -161,6 +161,7 @@ iceberg-aws = { group = "org.apache.iceberg", name = "iceberg-aws", version.ref
 iceberg-core = { group = "org.apache.iceberg", name = "iceberg-core", version.ref = "iceberg" }
 iceberg-api = { group = "org.apache.iceberg", name = "iceberg-api", version.ref = "iceberg" }
 iceberg-hive-metastore = { group = "org.apache.iceberg", name = "iceberg-hive-metastore", version.ref = "iceberg" }
+iceberg-gcp = { group = "org.apache.iceberg", name = "iceberg-gcp", version.ref = "iceberg" }
```
Comment (diqiu50) [inline gradle/libs.versions.toml:164]: Should this be added. if uses want to use, it's add the jars to class path himself

**Comment by jerryshao** [comment]
I'm OK with the change here. But it is not a good design to support adding cloud storage one by one continuously. We should figure out a framework to support adding by users.

File: docs/lakehouse-iceberg-catalog.md [hunk 2]
```diff
@@ -103,6 +103,22 @@ For other Iceberg OSS properties not managed by Gravitino like `client.security-
...
+
+Supports using google credential file to access GCS data.
+
+| Configuration item     | Description                                                                                        | Default value | Required | Since Version |
+|------------------------|----------------------------------------------------------------------------------------------------|---------------|----------|---------------|
+| `io-impl`              | The io implementation for `FileIO` in Iceberg, use `org.apache.iceberg.gcp.gcs.GCSFileIO` for GCS. | (none)        | No       | 0.6.0         |
+
+For other Iceberg GCS properties not managed by Gravitino like `gcs.project-id`, you could config it directly by `gravitino.bypass.gcs.project-id`.
+
+Please make sure the credential file is accessible by Gravitino, like using `export GOOGLE_APPLICATION_CREDENTIALS=/xx/application_default_credentials.json` before Gravitino server is started.
```
Comment (jerryshao) [inline docs/lakehouse-iceberg-catalog.md:116]: Does GCS need configurations like access key ?

**Comment by FANNG1** [comment]
> �I'm OK with the change here. But it is not a good design to support adding cloud storage one by one continuously. We should figure out a framework to support adding by users.

For more general storage, we could define properties explicitly, for others use #4896 to support.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**